control/grpc.py

```diff
@@ -602,6 +602,28 @@ def delete_subsystem(self, request, context=None):
                 self.logger.warning(f"Will continue deleting {request.subsystem_nqn} anyway")
         return self.execute_grpc_function(self.delete_subsystem_safe, request, context)
 
+    def check_if_image_used(self, pool_name, image_name):
+        """Check if image is used by any other namespace."""
+
+        errmsg = ""
+        nqn = None
+        state = self.gateway_state.local.get_state()
+        for key, val in state.items():
+            if not key.startswith(self.gateway_state.local.NAMESPACE_PREFIX):
+                continue
+            try:
+                ns = json.loads(val)
+                ns_pool = ns["rbd_pool_name"]
+                ns_image = ns["rbd_image_name"]
+                if pool_name and pool_name == ns_pool and image_name and image_name == ns_image:
+                    nqn = ns["subsystem_nqn"]
+                    errmsg = f"RBD image {ns_pool}/{ns_image} is already used by a namespace in subsystem {nqn}"
+                    break
+            except Exception:
```

Comment:
1. if we are expecting JSON Exception, maybe we should handle that specific exception type
2. not sure why the exception is just ignored here, looks like a fatal error to me.

Reply:
I didn't want to fail as this function is not critical. In the worst case we will not know that the image is used which is what we do now anyway.